### DISCUSSION TITLE
Could header mutator serialize nested values to json instead of go map?

### DISCUSSION THREAD

**Discussion by nmapx** [ory/oathkeeper#1142] [Ideas]
Technically it could be possible to use eg. X-User header containing all of the user profile info as JSON (just like it's being stored in metadata_public in Kratos) instead of passing each value in a separate header which might be hard to maintain. Currently if I use limited `print` or `printf "%s"` func on a nested object (in oathkeeper config), it is being serialized to Go map (as a string) which kinda works (i can see the values) but can't make it JSON formatted. So I was wondering - would you be open for such a possibility? If this kind of feature is not in your backlog I could possibly make a contribution.

**Comment by nmapx** [comment]
@vinckr Could you please have a look? I asked it couple months ago and found a workaround already but still it's not ideal and I'm open for any improvements in that matter.

**Reply by vinckr** [reply]
Hello @nmapx 
Apologies for the late answer. 
To be honest I am not working on Oathkeeper directly so I can't really say if this would be a good feature. 
Can you maybe use the [Feature Request template](https://github.com/ory/oathkeeper/issues/new?assignees=&labels=feat&projects=&template=FEATURE-REQUEST.yml) to create a bit more detailed feature request issue and then the core maintainers will take a look when they have time.
